cpp/include/proton/uuid.hpp

```diff
@@ -35,13 +38,20 @@ namespace proton {
 
 /// A 16-byte universally unique identifier.
 class uuid : public byte_array<16> {
+
+  private:
+    thread_local static std::independent_bits_engine<std::mt19937, CHAR_BIT,
+                                                     unsigned int>
+        engine;
+    thread_local static std::seed_seq seed;
+
```

Comment:
Please don't add these private members to the class. It changes the ABI (also the API but in a more subtle way). 
Since these are thread_locals anyway they don't need to be per instance variables so can just be static to the implementation file itself and not be in the class at all.

Reply:
@DreamPearl Right, ABI. Sorry I forgot to think about that before.

Reply:
Actually thinking a bit more. It probably doesn't actually change the ABI. But it does unnecessarily leak implementation details of the uuid class. IMO it's better to just make these static thread_local in the implementation file without putting them in the class at all. This will generate pretty much exactly the same code but is better encapsulated (again IMO)

Reply:
Moved them out of the class and put them all in struct construct.